crypto/keys/types.go

```diff
@@ -0,0 +1,218 @@
+package keys
+
+import (
+	"fmt"
+
+	emintCrypto "github.com/cosmos/ethermint/crypto"
+	"github.com/tendermint/tendermint/crypto"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/hd"
+	"github.com/cosmos/cosmos-sdk/types"
+)
+
+// Keybase exposes operations on a generic keystore
+type Keybase interface {
```

Comment:
Is there any benefit to using the existing interface defined in the SDK? As opposed to re-defining it here

Reply:
The rationale was because the types of certain functions, including CreateAccount have different types as the ones declared in the base Cosmos SDK to achieve the functionality needed (I had to change the Info type). The types are intended to be completely separate from the standard key structs/interfaces but I had just used and shared functionality when possible to reduce the amount of duplicate code and functionality needed. 

I know this doesn't come out very clean and I really meant to open this as a draft PR to find the cleanest solution without having to redeclare and rewrite all of this logic just to change the underlying key type and functionality.

Reply:
Upon looking at it again, it appears as if it may be possible to revert the Keybase and Info interfaces to be based on the cosmos SDK ones, and I will check if this is the case tomorrow or remember why I had done this previously.